terraso_backend/apps/soil_id/graphql/soil_id/resolvers.py

```diff
@@ -0,0 +1,272 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see https://www.gnu.org/licenses/.
+
+from typing import Optional
+
+from soil_id.us_soil import list_soils, rank_soils
+
+from apps.soil_id.graphql.soil_data import DepthInterval
+from apps.soil_id.graphql.soil_id.sample_data import (
+    dummy_data_matches,
+    dummy_location_matches,
+)
+from apps.soil_id.graphql.soil_id.schema import (
+    DataBasedSoilMatch,
+    DataBasedSoilMatches,
+    EcologicalSite,
+    LABColorInput,
+    LandCapabilityClass,
+    LocationBasedSoilMatch,
+    LocationBasedSoilMatches,
+    SoilIdDepthDependentData,
+    SoilIdInputData,
+    SoilIdSoilData,
+    SoilInfo,
+    SoilMatchInfo,
+    SoilSeries,
+)
+from apps.soil_id.models.depth_dependent_soil_data import DepthDependentSoilData
+from apps.soil_id.models.soil_data import SoilData
+
+
+def resolve_texture(texture: str | float):
+    if not isinstance(texture, str) or texture == "":
+        return None
+    return texture.upper().replace(" ", "_")
+
+
+def resolve_rock_fragment_volume(rock_fragment_volume: int | float | str):
+    if not (isinstance(rock_fragment_volume, float) or isinstance(rock_fragment_volume, int)):
+        return None
+    elif rock_fragment_volume <= 1:
+        return DepthDependentSoilData.RockFragmentVolume.VOLUME_0_1.value
+    elif rock_fragment_volume <= 15:
+        return DepthDependentSoilData.RockFragmentVolume.VOLUME_1_15.value
+    elif rock_fragment_volume <= 35:
+        return DepthDependentSoilData.RockFragmentVolume.VOLUME_15_35.value
+    elif rock_fragment_volume <= 60:
+        return DepthDependentSoilData.RockFragmentVolume.VOLUME_35_60.value
+    else:
+        return DepthDependentSoilData.RockFragmentVolume.VOLUME_60.value
+
+
+def resolve_soil_data(soil_match) -> SoilIdSoilData:
+    bottom_depths = soil_match["bottom_depth"]
+    prev_depth = 0
+    depth_dependent_data = [None] * len(bottom_depths)
+
+    for id, bottom_depth in bottom_depths.items():
+        depth_dependent_data[int(id)] = SoilIdDepthDependentData(
+            depth_interval=DepthInterval(start=prev_depth, end=bottom_depth),
+            texture=resolve_texture(soil_match["texture"][id]),
+            rock_fragment_volume=resolve_rock_fragment_volume(soil_match["rock_fragments"][id]),
+            munsell_color_string=soil_match["munsell"][id],
+        )
+        prev_depth = bottom_depth
+
+    return SoilIdSoilData(
+        slope=soil_match["site"]["siteData"]["slope"], depth_dependent_data=depth_dependent_data
+    )
+
+
+def resolve_ecological_site(ecological_site: dict):
+    if ecological_site["ecoclassid"] == "" or ecological_site["ecoclassid"][0] == "":
```

Comment:
```suggestion
    if not ecological_site["ecoclassid"] or not ecological_site["ecoclassid"][0]:
```

Reply:
i disagree that not being explicit about what we're testing for is better here